src/lua/LuaPiGui.cpp

```diff
@@ -493,6 +493,55 @@ static int l_pigui_get_scroll_y(lua_State *l)
 	return 1;
 }
 
+/*
+ * Function: plotHistogram
+ *
+ * Make a histogram
+ *
+ * > ui.plotHistogram(label, data, display_count, overlay, y_min, y_max, graph_size)
+ *
+ * Example:
+ *
+ * Plots 5 bars, with height 17,3,15,5, and 8, respectively
+ *
+ * > ui.plotHistogram("Test", {17,3,15,5,8}, 0)
+ *
+ * Parameters:
+ *
+ *   label - string, text on button
+ *   data - table of values
+ *   display_count - number of pionts to plot, 0 for all
+ *   overlay_text - optional title string, to put on histogram
+ *   y_min - optional float, setting min y-value displayed
+ *   y_max - optional float, setting max y-value displayed
+ *   graph_size - optional Vector2, defining size
+ *
+ */
+static int l_pigui_plot_histogram(lua_State *l)
+{
+	PROFILE_SCOPED()
+	std::string label = LuaPull<std::string>(l, 1);
+	LuaTable vals = LuaTable(l, 2);
+	float values[vals.Size()];
+	float max = FLT_MIN;
+	float min = FLT_MAX;
+	for (int i = 1; i <= vals.Size(); i++) {
+		values[i - 1] = vals.Get<int>(i);
+		if (values[i - 1] > max)
+			max = values[i - 1];
+		if (values[i - 1] < min)
+			min = values[i - 1];
+	}
+	int display_count = LuaPull<int>(l, 3);
```

Comment:
Ideally the default behavior should be to display all values in the data array; but we offer the caller more control as to how many values they'd like to display.

```suggestion
	int display_count = LuaPull<int>(l, 3, vals.Size());
```

Reply:
I'm not sure that variable does what I think it does. From Imgui demo example:

Here, it's called `display_count`:
``` c++
ImGui::PlotLines("Lines", func, NULL, display_count, 0, NULL, -1.0f, 1.0f, ImVec2(0,80));
ImGui::PlotHistogram("Histogram", func, NULL, display_count, 0, NULL, -1.0f, 1.0f, ImVec2(0,80));
```

Here it's called `values_offset`:

```c++
ImGui::PlotLines("Lines", values, IM_ARRAYSIZE(values), values_offset, overlay, -1.0f, 1.0f, ImVec2(0,80));
ImGui::PlotHistogram("Histogram", arr, IM_ARRAYSIZE(arr), 0, NULL, 0.0f, 1.0f, ImVec2(0,80));

```
Here's an experiment I did:
![2020-03-14-104449_1600x1800_scrot](https://user-images.githubusercontent.com/619390/76679632-6a615780-65e2-11ea-8016-b41a805a8889.png)

And in the screenshot above, explicitly setting it to 10, looks to just shift the plot (still plots 31 points), so it's an offset? Then we should probably expose the third argument on c++ side as well, assuming that is the _real_ display_count? 

(I'll test that now)